vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -407,7 +407,12 @@ void ReactInstanceWin::OnReactInstanceLoaded(const Mso::ErrorCode &errorCode) no
       if (auto strongThis = weakThis.GetStrongPtr()) {
         if (!strongThis->m_isLoaded) {
           strongThis->m_isLoaded = true;
-          strongThis->m_state = ReactInstanceState::Loaded;
+          if (!errorCode) {
```

Comment:
when I looked at this, I saw that if I let the instance stay in HasError, I could never reload the app:
the repro was:
don't start the bundler
start the app, get redbox, instance goes to HasError
press reload ----> no reload happens but redbox gets dismissed. app is left unusable.

Can you please confirm this works? can you think of why I might have seen that? 😋

Reply:
The reload button works. If I dismiss the redbox, the app seems to unusable, but I can click Ctrl+Shift+D to reload it. If I run the "yarn start" before the reload, then I see the UI.

I guess the main difference between this change and what probably you did is that I let all other code to run as it was before. It is very important for the rest of the system to report that the load process is completed, even if it is completed in an error state. It allows to propagate all the messages and complete the load operation in the ActionQueue.

The ActionQueue expects that each action reports completion before it starts the next one. BTW, it also coalesce the actions. E.g. if unload is requested multiple times, only one of them is kept.

Reply:
I have verified the scenario again.  The reloading worked fine as described above. I saw the app crash when we resize the window. It was caused by the RedBox code. I think we must clean the m_redboxContent on the RedBox closing.

@acoates-ms could you please see if changes to the RedBox look good?